### PR TITLE
Support disabling training kernels as part of a reduced build

### PR DESCRIPTION
**Pull request by cianmhayes**
**Description**: Include training kernel registration in the set of kernel registration files

**Motivation and Context**
When building with training support and a reduced kernel set, the unused training kernels aren't disabled.

### CODE REVIEW

**Comment by mrry** [comment]
This looks good to me, but I'll leave it to @skottmckay to approve (he'll know if there are any gotchas).

**Comment by cianmhayes** [comment]
@skottmckay , would you mind taking a look at this change?

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**
